internal/server/controller/grpc/delay.go

```diff
@@ -0,0 +1,83 @@
+package grpc
+
+import (
+	"context"
+	"time"
+
+	pb "github.com/Liar233/cronos-shovel/internal/pkg"
+	"github.com/Liar233/cronos-shovel/internal/server/model"
+	"github.com/Liar233/cronos-shovel/internal/server/storage/repository"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+type DelayController struct {
+	pb.UnimplementedDelayControllerServer
+	logger    logrus.FieldLogger
+	delayRepo repository.DelayRepositoryInterface
+}
+
+func NewDelayController(
+	logger logrus.FieldLogger,
+	delayRepo repository.DelayRepositoryInterface,
+) *DelayController {
+
+	return &DelayController{
+		logger:    logger,
+		delayRepo: delayRepo,
+	}
+}
+
+func (dc *DelayController) CreateDelay(ctx context.Context, req *pb.CreateDelayRequest) (*pb.CreateDelayResponse, error) {
+
+	resp := &pb.CreateDelayResponse{}
+
+	messageId, err := uuid.Parse(req.MessageId)
+
+	if err != nil {
+		resp.Error = err.Error()
+		dc.logger.Error("Invalid delay message_id: %s", err.Error())
+
+		return resp, err
```

Comment:
```suggestion
		return nil, status.Error(codes.Internal, msg)
```

Только нужно обработать ошибки, чтобы вернуть нужный код

Reply:
+